libs/framework/src/celix_launcher.c

```diff
@@ -34,135 +34,130 @@
+		celix_properties_destroy(embeddedConfig);
 		return 0;
 	}
 
+    if (createCache) {
+        return celixLauncher_createBundleCache(embeddedConfig, configFile);
+    }
+
     if (showEmbeddedBundles) {
-        printEmbeddedBundles();
-        celix_properties_destroy(packedConfig);
+        celixLauncher_printEmbeddedBundles();
+        celix_properties_destroy(embeddedConfig);
         return 0;
     }
 
 	struct sigaction sigact;
 	memset(&sigact, 0, sizeof(sigact));
-	sigact.sa_handler = shutdown_framework;
+	sigact.sa_handler = celixLauncher_shutdownFramework;
 	sigaction(SIGINT,  &sigact, NULL);
 	sigaction(SIGTERM, &sigact, NULL);
 
 	memset(&sigact, 0, sizeof(sigact));
-	sigact.sa_handler = ignore;
+	sigact.sa_handler = celixLauncher_ignore;
 	sigaction(SIGUSR1,  &sigact, NULL);
 	sigaction(SIGUSR2,  &sigact, NULL);
 
 
-	int rc = celixLauncher_launchWithConfigAndProps(config_file, &framework, packedConfig);
+	int rc = celixLauncher_launchWithConfigAndProps(configFile, &framework, embeddedConfig);
 	if (rc == 0) {
 		g_fw = framework;
-		celixLauncher_waitForShutdown(framework);
-		celixLauncher_destroy(framework);
+		*frameworkOut = framework;
 	}
 	return rc;
 }
 
-static void show_usage(char* prog_name) {
-	printf("Usage:\n  %s [-h|-p] [path/to/runtime/config.properties]\n", basename(prog_name));
-	printf("Options:\n");
-	printf("\t-h | --help: Show this message\n");
-	printf("\t-p | --props: Show the embedded and runtime properties for this Celix container\n");
-    printf("\t--embedded_bundles: Show the embedded bundles for this Celix container\n");
-	printf("\n");
-}
-
-static void shutdown_framework(int signal) {
+static void celixLauncher_shutdownFramework(int signal) {
 	if (g_fw != NULL) {
 		celixLauncher_stop(g_fw); //NOTE main thread will destroy
 	}
 }
 
-static void ignore(int signal) {
+static void celixLauncher_ignore(int signal) {
 	//ignoring for signal SIGUSR1, SIGUSR2. Can be used on threads
 }
 
 int celixLauncher_launch(const char *configFile, celix_framework_t* *framework) {
 	return celixLauncher_launchWithConfigAndProps(configFile, framework, NULL);
 }
 
-static int celixLauncher_launchWithConfigAndProps(const char *configFile, celix_framework_t* *framework, celix_properties_t* packedConfig) {
-	if (packedConfig == NULL) {
-		packedConfig = celix_properties_create();
-	}
-
-	FILE *config = fopen(configFile, "r");
-	if (config != NULL) {
-		celix_properties_t *configProps = celix_properties_loadWithStream(config);
-		fclose(config);
-		combine_properties(packedConfig, configProps);
-		celix_properties_destroy(configProps);
-	}
-
-	return celixLauncher_launchWithProperties(packedConfig, framework);
+static int celixLauncher_launchWithConfigAndProps(const char *configFile, celix_framework_t** framework, celix_properties_t* packedConfig) {
+    celix_properties_t* config = celixLauncher_createConfig(configFile, packedConfig);
+    return celixLauncher_launchWithProperties(config, framework);
```

Comment:
`CELIX_ILLEGAL_STATE` is bad for exit status:

> Although any value in the range 0 to 255 can be passed to the parent via the
status argument to _exit(), specifying values greater than 128 can cause confu-
sion in shell scripts. The reason is that, when a command is terminated by a
signal, the shell indicates this fact by setting the value of the variable $? to 128
plus the signal number, and this value is indistinguishable from that yielded
when a process calls _exit() with the same status value.

Reply:
yeah good catch. Updated to not return celix status values.